spec/requests/chat_channel_memberships_spec.rb

```diff
@@ -10,90 +10,295 @@
+        end.to raise_error(ActiveRecord::RecordNotFound)
+      end
+    end
+  end
+
+  describe "GET /chat_channel_memberships/:id/edit" do
+    before do
+      chat_channel.add_users([second_user])
+    end
+
+    let(:chat_channel_membership) { chat_channel.chat_channel_memberships.where(user_id: second_user.id).first }
+
+    context "when user is not logged in" do
+      it "raise Pundit::NotAuthorizedError" do
+        expect do
+          get "/chat_channel_memberships/#{chat_channel_membership.id}/edit"
+        end.to raise_error(Pundit::NotAuthorizedError)
+      end
+    end
+
+    context "when user is logged in and channel id is wrong" do
+      it "raise ActiveRecord::RecordNotFound" do
+        sign_in second_user
+        expect do
+          get "/chat_channel_memberships/ERW/edit"
+        end.to raise_error(ActiveRecord::RecordNotFound)
+      end
+    end
+
+    context "when user is a channel member" do
+      it "allows user to view channel members" do
+        sign_in second_user
+        get "/chat_channel_memberships/#{chat_channel_membership.id}/edit"
+        expect(response.body).to include("Members")
+        expect(response.body).to include(user.username.to_s)
+        expect(response.body).to include(second_user.username.to_s)
+        expect(response.body).not_to include("Pending Invitations")
+      end
+    end
+
+    context "when user is a channel moderator" do
+      it "allows user to view channel members" do
+        sign_in second_user
+        chat_channel_membership.update(role: "mod")
+        get "/chat_channel_memberships/#{chat_channel_membership.id}/edit"
+        expect(response.body).to include("Members")
+        expect(response.body).to include(user.username.to_s)
+        expect(response.body).to include(second_user.username.to_s)
+        expect(response.body).to include("Pending Invitations")
+        expect(response.body).to include("You are a channel mod")
+      end
+    end
+  end
+
   describe "POST /chat_channel_memberships" do
-    it "creates chat channel invitation" do
-      user.add_role(:super_admin)
-      mems_num = ChatChannelMembership.all.size
-      post "/chat_channel_memberships", params: {
-        chat_channel_membership: {
-          user_id: second_user.id, chat_channel_id: chat_channel.id
+    context "when user is super admin" do
+      it "creates chat channel invitation" do
+        user.add_role(:super_admin)
+        chat_channel_members_count = ChatChannelMembership.all.size
+        post "/chat_channel_memberships", params: {
+          chat_channel_membership: {
+            invitation_usernames: second_user.username.to_s,
+            chat_channel_id: chat_channel.id
+          }
         }
-      }
-      expect(ChatChannelMembership.all.size).to eq(mems_num + 1)
-      expect(ChatChannelMembership.last.status).to eq("pending")
+        expect(ChatChannelMembership.all.size).to eq(chat_channel_members_count + 1)
```

Comment:
I suppose rspec [`expect {}.to change(object, :count)`](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher) could be used here.